### PR TITLE
Added shared embedding option to director model.

### PR DESCRIPTION
**Pull request by leox1v**
Added an option (with flag --director-use-shared-embedding) to share the generation and classification heads of the director. Only two additional parameters are introduced to scale the logits before applying the sigmoid (for the classification).

### CODE REVIEW

**Review by jaseweston — APPROVED** [review]

File: projects/director/director_agent.py [hunk 1]
```diff
@@ -56,6 +75,9 @@ def classifier_output(self, input: torch.Tensor):
         if self.freeze_decoder:
             input = input.detach()
 
+        if self.use_shared_embedding:
```
Comment (jxmsML) [inline projects/director/director_agent.py:78]: should this come before the L75 `input = input.detach()`

File: projects/director/director_agent.py [hunk 2]
```diff
@@ -26,19 +26,38 @@
...
+        use_shared_embedding=False,
+        **kwargs,
+    ):
         super().__init__(opt, dictionary, **kwargs)
 
         vocabulary_size = len(dictionary)
 
         decoder_output_dim = self.decoder.out_dim
-        self.classifier_heads = nn.Linear(decoder_output_dim, vocabulary_size)
+        self.use_shared_embedding = use_shared_embedding
```
Comment (jxmsML) [inline projects/director/director_agent.py:56]: if `use_shared_embedding` is passing from opt, then you can make sure of that  `self.use_shared_embedding = opt.get('use_shared_embedding', False)?

Comment (leox1v) [reply projects/director/director_agent.py]: Thanks for the comment. I changed it.

**Review by jxmsML — APPROVED** [review]
